txscript/sign.go

```diff
@@ -557,14 +513,14 @@ func mergeScripts(chainParams dcrutil.AddressParams, tx *wire.MsgTx, idx int,
 // KeyDB is an interface type provided to SignTxOutput, it encapsulates
 // any user state required to get the private keys for an address.
 type KeyDB interface {
-	GetKey(dcrutil.Address) (chainec.PrivateKey, bool, error)
+	GetKey(dcrutil.Address) ([]byte, bool, error)
 }
 
 // KeyClosure implements KeyDB with a closure.
-type KeyClosure func(dcrutil.Address) (chainec.PrivateKey, bool, error)
+type KeyClosure func(dcrutil.Address) ([]byte, bool, error)
```

Comment:
This seems like it should return the type as well based on the address so it can be used in `RawTxInSignature` instead of hard coding the type which would be wrong for other types of addresses.